@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,190 @@ export namespace MapViewUtils {
+        // value stays constant when tilting but using the same X & Y offsets. We use the triangle
+        // with these two vectors from the camera position and the intersection of the flat plane.
+        // We know two angles, and the third is easy to compute (just 180 minus the sum of the
+        // rest). the first angle is 90 degrees + the current tilt, i.e. the tilt at the map
+        // center. The second angle is 90 - the tilt at the rotation center. Note, this doesn't work
+        // for sphere, it would have to be fixed for that.
+        const angleBetweenRotationTargetAndMapTarget =
+            ninetyRad * 2 -
+            (ninetyRad + angleBetweenNormals - currentTiltRotationCenter + ninetyRad + currentTilt);
+
+        // This uses the same formula as above, just we change `currentTilt` to the maxTiltAngle,
+        // i.e we want to find the greatest angle at the rotation target that gives us the max
+        // angle at the map center target.
+        const maxRotationTargetAngle =
+            ninetyRad * 2 - angleBetweenRotationTargetAndMapTarget - ninetyRad - maxTiltAngle;
+
+        // We compute the max tilt based on the angle of the inner triangle, note, for sphere
+        // projection this is just an approximation, in order to compute this exactly, we would
+        // need to shift the camera to the new position, compute the new normal at the map center
+        // and use this to then compute the angleBetweenNormals, it would then actually need to be
+        // done iteratively, because when shifting the camera, it is possible that we move it to a
+        // position which is outside the max tilt given the new normal. Because the `deltaTilt` is
+        // generally small, the difference in normals should be for most cases very small and the
+        // extra computation required to compute such a small value isn't worth it. The value we
+        // compute is anyway conservative, so we don't risk computing something that will put the
+        // camera outside the valid range.
+        const maxTilt = ninetyRad + angleBetweenNormals - maxRotationTargetAngle;
+
+        const clampedDeltaTiltMapTarget =
+            MathUtils.clamp(deltaTilt + currentTiltRotationCenter, 0, maxTilt) -
+            currentTiltRotationCenter;
+
+        const maxRotationTiltAngle = THREE.MathUtils.degToRad(89);
+        // Here we clamp to 89 degrees, to prevent it intersecting with the world at 90, note, we
+        // don't use the `maxTiltAngle`, because then special handling needs to be applied when for
+        // example the tilt at the rotation center is more than that at the map target, then the
+        // tilt feels different at the center / offset positions. Using 89 gives the most natual
+        // feeling when applying the tilt angle.
+        const clampedDeltaTiltRotationTarget =
+            MathUtils.clamp(deltaTilt + currentTiltRotationCenter, 0, maxRotationTiltAngle) -
+            currentTiltRotationCenter;
```

Comment:
after this: 
clampedDeltaTiltRotationTarget <= (89 - currentTiltRotationCenter)